### PR TITLE
Character classes

### PR DESCRIPTION
**Pull request by EhsanShahbazii**
/done

### CODE REVIEW

**Comment by EhsanShahbazii** [comment]
I apologize for my delay.

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 1]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
 
-Also we can use many character classes:
+همچنین می توانیم از بسیاری از کلاس های کاراکتر استفاده کنیم:
 
 ```js run
 alert( "I love HTML5!".match(/\s\w\w\w\w\d/) ); // ' HTML5'
 ```
 
-The match (each regexp character class has the corresponding result character):
+مطابقت (هر کلاس کاراکتر regexp دارای کاراکتر نتیجه مربوطه است):
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:68]:
```suggestion
عبارت دارای مطابقت (هر کلاس کاراکتر regexp دارای کاراکتر نتیجه مربوطه است):
```

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 2]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
 ```
 
-The pattern `pattern:[\s\S]` literally says: "a space character OR not a space character". In other words, "anything". We could use another pair of complementary classes, such as `pattern:[\d\D]`, that doesn't matter. Or even the `pattern:[^]` -- as it means match any character except nothing.
+الگوی `pattern:[\s\S]` به معنای واقعی کلمه می گوید: "یا کاراکتر فاصله یا نه". به عبارت دیگر، "هر چیزی". می‌ توانیم از یک جفت کلاس مکمل دیگر، مانند `pattern:[\d\D]` استفاده کنیم، که مهم نیست. یا حتی `[^]:pattern` -- به این معنی که با هر کاراکتری به جز هیچ چیز مطابقت دارد.
 
-Also we can use this trick if we want both kind of "dots" in the same pattern: the actual dot `pattern:.` behaving the regular way ("not including a newline"), and also a way to match "any character" with `pattern:[\s\S]` or alike.
+همچنین اگر بخواهیم هر دو نوع `نقطه` را در یک الگوی یکسان بخواهیم، می‌ توانیم از این ترفند استفاده کنیم: `.:pattern` نقطه واقعی که به روش معمولی رفتار می‌کند (خط جدید را شامل نمی شود)، و همچنین راهی برای مطابقت با "هر کاراکتری" با `pattern:[\s\S]` یا مشابه.
 ````
 
 ````warn header="Pay attention to spaces"
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:161]:
```suggestion
````warn header="به فاصله ها توجه کنید"
```
Headers have to be translated.

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 3]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
 
-Let's fix it adding spaces into the regexp `pattern:\d - \d`:
+بیایید با افزودن فاصله به regexp به صورت `pattern:\d - \d آن را برطرف کنیم:
 
 ```js run
-alert( "1 - 5".match(/\d - \d/) ); // 1 - 5, now it works
-// or we can use \s class:
-alert( "1 - 5".match(/\d\s-\s\d/) ); // 1 - 5, also works
+alert( "1 - 5".match(/\d - \d/) ); // 1 - 5, حال درست کار می کند
+// یا می توانیم از کلاس s\ استفاده کنیم:
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:176]:
```suggestion
// :استفاده کنیم \s یا می توانیم از کلاس
```
Please follow this [guide](https://github.com/javascript-tutorial/fa.javascript.info#%D8%AA%D8%B1%D8%AC%D9%85%DB%80-%DA%A9%D8%A7%D9%85%D9%86%D8%AA%D9%87%D8%A7-%D8%AF%D8%B1-%DA%A9%D8%AF) for translating comments.

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 4]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
 
-**A space is a character. Equal in importance with any other character.**
+**فضا یک کاراکتر است. از نظر اهمیت با هر کاراکتر دیگری برابر است.**
 
-We can't add or remove spaces from a regular expression and expect it to work the same.
 
-In other words, in a regular expression all characters matter, spaces too.
+ما نمی‌ توانیم فاصله‌ها را از یک عبارت باقاعده اضافه یا حذف کنیم و انتظار داشته باشیم که به همان صورت عمل کند.
+
+به عبارت دیگر، در یک عبارت باقاعده، همه کاراکترها مهم هستند. فضاها نیز همین طور.
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:185]:
```suggestion
به عبارت دیگر، در یک عبارت باقاعده، همه کاراکترها مهم هستند. فاصله ها نیز همین طور.
```

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 5]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
-There exist following character classes:
+کلاس های کاراکتر زیر وجود دارد:
 
-- `pattern:\d` -- digits.
-- `pattern:\D` -- non-digits.
-- `pattern:\s` -- space symbols, tabs, newlines.
+- `pattern:\d` -- رقمی.
+- `pattern:\D` -- غیر رقمی.
+- `pattern:\s` -- نمادهای فاصله، تب ها، خطوط جدید.
 - `pattern:\S` -- all but `pattern:\s`.
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:195]:
```suggestion
- `pattern:\S` -- هر چیزی به جز `pattern:\s`.
```

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 6]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
-- `pattern:\w` -- Latin letters, digits, underscore `'_'`.
+- `pattern:\w` -- حروف لاتین، اعداد، خط زیر `'_'`.
 - `pattern:\W` -- all but `pattern:\w`.
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:197]:
```suggestion
- `pattern:\W` -- هر چیزی به جز `pattern:\w`.
```

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 7]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
-- `pattern:.` -- any character if with the regexp `'s'` flag, otherwise any except a newline `\n`.
+- `pattern:.` -- هر کاراکتری اگر `regxp` با پرچم `'s'`، در غیر این صورت هر کاراکتری به جز خط جدید `n\`.
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:198]:
```suggestion
- `pattern:.` -- اگر با پرچم `'s'` همراه باشد هر کاراکتری مورد قبول است، در غیر این صورت هر کاراکتری به جز خط جدید `n\`.
```

File: 9-regular-expressions/02-regexp-character-classes/article.md [hunk 8]
```diff
@@ -59,145 +59,146 @@ let regexp = /CSS\d/
...
 
-...But that's not all!
+...اما این همه ماجرا نیست!
 
-Unicode encoding, used by JavaScript for strings, provides many properties for characters, like: which language the letter belongs to (if it's a letter), is it a punctuation sign, etc.
+رمزگذاری یونیکد، که توسط جاوا اسکریپت برای رشته ها استفاده می شود، ویژگی های بسیاری را برای کاراکترها فراهم می کند، مانند: حروف به کدام زبان تعلق دارد (اگر حرف باشد)، آیا علامت نقطه گذاری است و غیره.
```
Comment (mahdiHash) [inline 9-regular-expressions/02-regexp-character-classes/article.md:202]:
```suggestion
رمزگذاری یونیکد، که توسط جاوااسکریپت برای رشته ها استفاده می شود، ویژگی های بسیاری را برای کاراکترها فراهم می کند، مانند: حروف به کدام زبان تعلق دارد (اگر حرف باشد)، آیا علامت نقطه گذاری است و غیره.
```

**Review by mahdiHash — CHANGES_REQUESTED** [review]
Please review the changes.
You can add the suggestions to a batch and submit a single commit ([docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes)).

You inserted an empty line 182, please remove it.

**Comment by mahdiHash** [comment]
> I apologize for my delay.

No worries.

**Comment by mahdiHash** [comment]
Thanks for your contribution, @EhsanShahbazii.

**Pull request merged**